### PR TITLE
Add SSL connection to irc

### PR DESCRIPTION
**Pull request by Nickwasused**
# Twitch
Twitch Offers both an unencrypted connection and an encrypted connection to their IRC servers.
The unencrypted Connection is to Port 6667 and the encrypted is to Port 6697.

(See more about that here: https://dev.twitch.tv/docs/irc/guide)

# Implementation
I have added a Settings option for Choosing the connection type.

# Other Stuff
The Chat Server Url has changed from "irc.twitch.tv" to "irc.chat.twitch.tv" as seen here: https://dev.twitch.tv/docs/irc/guide.

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -275,6 +275,7 @@
     <!-- Changelog -->
     <string name="changelog_title">Changelog</string>
     <string name="changelog_show_after_update">Show changelog after an update</string>
+    <string name="chat_enable_ssl">Enable Secure Connection</string>
```
Comment (samfundev) [inline app/src/main/res/values/strings.xml:278]: This should go under the `Chat Settings` section and use sentence case like the other settings.

File: app/src/main/java/com/perflyst/twire/service/Settings.java [hunk 2]
```diff
@@ -59,6 +59,7 @@
     private final String CHAT_LANDSCAPE_ENABLE = "chatLandscapeEnable";
     private final String CHAT_LANDSCAPE_SWIPEABLE = "chatLandscapeSwipable";
     private final String CHAT_LANDSCAPE_WIDTH = "chatLandscapeWidth";
+    private final String CHAT_ENABLE_SSL = "chatenablessl";
```
Comment (samfundev) [inline app/src/main/java/com/perflyst/twire/service/Settings.java:62]: Can you make the capitalization consistent?
```suggestion
    private final String CHAT_ENABLE_SSL = "chatEnableSSL";
```

File: app/src/main/java/com/perflyst/twire/chat/ChatManager.java [hunk 3]
```diff
@@ -186,8 +201,21 @@ protected void onPostExecute(Void aVoid) {
...
+            if (port != twitchChatPortsecure) {
+                socket = new Socket(address, port);
+            } else {
+                // if we use the SSL Port then create a SSL Socket
+                // https://stackoverflow.com/questions/13874387/create-app-with-sslsocket-java
+                SSLSocketFactory factory=(SSLSocketFactory) SSLSocketFactory.getDefault();
+                socket=(SSLSocket) factory.createSocket(address, port);
+            }
+
+            Log.d("is connected", String.valueOf(socket.isConnected()));
```
Comment (samfundev) [inline app/src/main/java/com/perflyst/twire/chat/ChatManager.java:217]: This and the next 2 logging lines (reader & Line) seem unnecessary, could you remove them if they're not needed?

**Review by samfundev — CHANGES_REQUESTED** [review]
Thanks for this PR! I did notice a few things that could be cleaned up before merging this in.

File: app/src/main/java/com/perflyst/twire/chat/ChatManager.java (hunk 3 above)
Comment (Nickwasused) [reply app/src/main/java/com/perflyst/twire/chat/ChatManager.java]: They have been removed in: https://github.com/twireapp/Twire/pull/226/commits/66a79c8ef2c86c3ff78a35e108785bc89679f95a

File: app/src/main/java/com/perflyst/twire/service/Settings.java (hunk 2 above)
Comment (Nickwasused) [reply app/src/main/java/com/perflyst/twire/service/Settings.java]: Fixed in: https://github.com/twireapp/Twire/pull/226/commits/66a79c8ef2c86c3ff78a35e108785bc89679f95a

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (Nickwasused) [reply app/src/main/res/values/strings.xml]: This has been changed in: https://github.com/twireapp/Twire/pull/226/commits/66a79c8ef2c86c3ff78a35e108785bc89679f95a

**Review by samfundev — APPROVED** [review]

**Comment by samfundev** [comment]
Everything looks good now, I'll merge this in.

**Pull request merged**